File: src/WePrint/Models/AddressModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Reflection;

namespace WePrint.Models
{
    public class AddressModel : DbModel
    {
        public string StreetAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public int ZipCode { get; set; }

        public void ApplyChanges(AddressUpdateModel update)
        {
            ReflectionHelper.CopyPropertiesTo(update, this);
        }
    }

    public class AddressUpdateModel
    {
        public string Id { get; set; }
        public string StreetAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public int? ZipCode { get; set; }
    }
}

File: src/WePrint/ClientApp/src/models/AddressModel.js
export default class AddressModel {
  constructor(streetAddress, city, state, zipCode) {
    this.streetAddress = streetAddress;
    this.city = city;
    this.state = state;
    this.zipCode = zipCode;
  }

  static AllPropsEqual(a, b) {
    if (a === b && a === null) return true;
    if (a !== null || b !== null) return false;
    if (a === b && a === undefined) return true;
    if (a !== undefined || b !== undefined) return false;
    return (
      a.streetAddress === b.streetAddress &&
      a.city === b.city &&
      a.state === b.state &&
      a.zipCode === b.zipCode
    );
  }
}
